aten/src/ATen/native/LinearAlgebra.cpp

```diff
@@ -1489,12 +1489,14 @@ static void addmm_impl_cpu_(
   // it is faster to call oneDNN matrix multiplication primitive with RHS*LHS
   // that will call then into Arm® Compute Library (ACL) GEMM kernel and also
   // additionally have support for running kernel with BF16 instructions
-  bool apply_heur = apply_mkldnn_matmul_heur(b.sizes()[0], b.sizes()[1], a.sizes()[1]);
-  if (apply_heur && transpose_a && !transpose_b && result.scalar_type() == at::ScalarType::Float) {
-      mkldnn_matmul(b, a, c, beta.to<float>(), alpha.to<float>());
-      // We have dispatched to ACL GEMM for single precision float
-      // so do not need to dispatch to BLAS GEMM below
-      dispatched = true;
+  if (transpose_c) {
```

Comment:
What's the problem of C non-transposed? Is it a limitation of oneDNN on AArch64? If so, it would be better to revise the comment above to elaborate on this.

Reply:
If C non-transposed, m1 and m2 will not be swapped (line 1430), thus should not to swap a and b, so we should call:
```c++
mkldnn_matmul(a, b, c, beta.to<float>(), alpha.to<float>());
```
but not:
```c++
mkldnn_matmul(b, a, c, beta.to<float>(), alpha.to<float>());
```
There is no problem with onednn. The problem is that the tensors passed are wrong.